tests/ui/const-generics/transmute-fail.stderr

```diff
@@ -10,8 +10,8 @@ error[E0512]: cannot transmute between types of different sizes, or dependently-
 LL |         std::mem::transmute(v)
    |         ^^^^^^^^^^^^^^^^^^^
    |
-   = note: source type: `[[u32; H + 1]; W]` (size can vary because of [u32; H + 1])
-   = note: target type: `[[u32; W + 1]; H]` (size can vary because of [u32; W + 1])
+   = note: source type: `[[u32; H + 1]; W]` (the type `[u32; H + 1]` has an unknown layout)
```

Comment:
This error still looks suspicious to me -- we should be reporting "size can vary" and not "unknown layout" here.

It looks like the errors come from here:
https://github.com/rust-lang/rust/blob/01a26c026ddd9f0da0307468b07b003b8759fc4a/compiler/rustc_middle/src/ty/layout.rs#L416-L419

But I'm not sure why this is constructing new `LayoutError::Unknown`s for some cases and returning the original layout error (`err`) for other cases.

Reply:
This is a little bit strange. 
Shoule we return ` LayoutError::TooGeneric` or the original error in `SizeSkeleton::Generic` case?

At line 416 by returning the original one does not cause any broken tests.

Reply:
Returning the original error should be fine (for both cases). If I'm following this code correctly, then `err` and `LayoutError::Unknown(ty)` before this PR / `LayoutError::TooGeneric(ty)` after this PR should always be the same anyway, so no reason to alloc a new error.

Reply:
I thought about the same thing. Now, the original error at line 419 should be `LayoutError::TooGeneric`.